src/main/java/org/terasology/launcher/platform/Platform.java

```diff
@@ -1,70 +1,87 @@
-// Copyright 2020 The Terasology Foundation
+// Copyright 2023 The Terasology Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-package org.terasology.launcher.util;
+package org.terasology.launcher.platform;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+import java.util.List;
+import java.util.Set;
 
 /**
  * A simplified representation of a computer platform as `os` and `arch`
  */
 public final class Platform {
+    public static final Set<Platform> SUPPORTED_PLATFORMS = Sets.newHashSet(
+            new Platform(OS.WINDOWS, Arch.X64),
+            new Platform(OS.LINUX, Arch.X64),
+            new Platform(OS.MAC, Arch.X64)
+    );
     
     private static final Platform PLATFORM = new Platform();
 
-    private String os;
-    private String arch;
+    private final OS os;
+    private final Arch arch;
 
     /**
      * Constructs platform information for the current host system.
      * Simplifies operating system name to one of `linux`, `mac`, `windows` if applicable.
      * Simplifies operating system architecture to one of `32` and `64` if applicable.
      */
-    private Platform() {
+    private Platform() throws UnsupportedPlatformException{
         final String platformOs = System.getProperty("os.name").toLowerCase();
         // TODO: consider using regex
         if (platformOs.startsWith("linux")) {
-            os = "linux";
+            os = OS.LINUX;
         } else if (platformOs.startsWith("mac os")) {
-            os = "mac";
+            os = OS.MAC;
         } else if (platformOs.startsWith("windows")) {
-            os = "windows";
+            os = OS.WINDOWS;
         } else {
-            os = platformOs;
+            throw new UnsupportedPlatformException();
         }
 
         final String platformArch = System.getProperty("os.arch");
         if (platformArch.equals("x86_64") || platformArch.equals("amd64")) {
-            arch = "64";
+            arch = Arch.X64;
         } else if (platformArch.equals("x86") || platformArch.equals("i386")) {
-            arch = "32";
+            arch = Arch.X86;
         } else {
-            arch = platformArch;
+            arch = Arch.X64;
+            //TODO: throw new UnsupportedPlatformException();
```

Comment:
Little hack for me to be able to develop on M1 Mac 😕  With this in place, the static platform initialisation will throw an exception and I cannot even run the tests.

I think this is kind of nice, because we can detect whether running the launcher would have make any sense on that platform or fail early on, but it is a bit annoying for development.

The brain split situation here is that the launcher can run on more platforms than the actual game. Maybe I'll add a "development mode" to be able to start the launcher on your own risk 🤔